packages/safe-tools-client/app/services/hub-authentication.ts

```diff
@@ -49,26 +48,36 @@ export default class HubAuthenticationService extends Service {
   }
 
   get authToken(): string | null {
-    return this.storage.getItem('authToken');
+    const authTokens = this.storage.getItem('authTokens');
+    if (!this.wallet.address || !authTokens) return null;
+    const authTokensObj = JSON.parse(authTokens);
+    return authTokensObj[this.wallet.address];
   }
 
   set authToken(val: string | null) {
     if (val) {
-      this.storage.setItem('authToken', val);
+      const authTokens = this.storage.getItem('authTokens');
+      const authTokensObj = authTokens ? JSON.parse(authTokens) : {};
```

Comment:
I think it's safer if we avoid stringifying and parsing an object and store individual entries instead

The keys could be `auth-token-${walletAddress}-${network}` for more clarity and easier debugging/inspection

Reply:
updated